spec/lib/contextual_logger/context_handler_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'contextual_logger'
+
+class ContextualLoggerContextHandlerSpecContainer
+  include ::ContextualLogger::Context
+end
+
+RSpec.describe ContextualLogger::ContextHandler do
+  let(:context) { { service: { name: 'tts', description: 'TTS' }, integration: "google" } }
+  let(:context2) { { service: { description: 'Context 2' }, integration: "google" } }
+
+  let(:instance) { ContextualLoggerContextHandlerSpecContainer.new }
+
+  subject(:handler) { described_class.new(instance, context) }
+
+  it { is_expected.to respond_to(:reset!) }
+
+  it 'resets the thread context on reset!' do
+    instance.current_context_override = context2
+    handler.reset!
+
+    expect(instance.current_context_override).to eq(context)
+  end
```

Comment:
⛏️  A slightly more "rspec" way to do this test

```suggestion
  describe '#reset!'
    before  { instance.current_context_override = context2 }

    it { expect { hander.reset! }.to change(instance, :current_context_override).from(context2).to(context) }
  end
```